mixnet/src/crypto.rs

```diff
@@ -64,3 +84,55 @@ impl From<sphinx_packet::crypto::PublicKey> for PublicKey {
         Self(public_key)
     }
 }
+
+impl Serialize for PublicKey {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        SerializablePublicKey::from(self).serialize(serializer)
+    }
+}
+
+impl<'de> Deserialize<'de> for PublicKey {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Ok(Self::from(SerializablePublicKey::deserialize(
+            deserializer,
+        )?))
+    }
+}
+
+// Only for serializing/deserializing [`PrivateKey`] since [`sphinx_packet::crypto::PrivateKey`] is not serializable.
+#[derive(Serialize, Deserialize, Clone, Debug)]
+struct SerializablePrivateKey([u8; 32]);
```

Comment:
Probably just deserialize.

Reply:
Just removed the `crypto.rs`: https://github.com/logos-co/nomos-node/pull/592#discussion_r1516857984